tool/planet/etcd.go

```diff
@@ -272,22 +271,30 @@ func etcdUpgrade(rollback bool) error {
 		}
 	}
 
-	// reset the kubernetes api server to take advantage of any new etcd settings that may have changed
-	// this only happens if the service is already running
-	status, err := getServiceStatus(APIServerServiceName)
-	if err != nil {
-		log.Warnf("Failed to query status of service %v. Continuing upgrade. Error: %v", APIServerServiceName, err)
-	} else {
-		if status != "inactive" {
-			tryResetService(ctx, APIServerServiceName)
-		}
-	}
-
 	log.Info("Upgrade complete")
 
 	return nil
 }
 
+// restartEtcdClients - because the etcd cluster has been recreated, all clients need to be refreshed so their
+// watches are not pointing at incorrect revisions.
+func restartEtcdClients(ctx context.Context) {
+	services := []string{APIServerServiceName, PlanetAgentServiceName, FlannelServiceName}
+
+	for _, service := range services {
+		// reset the kubernetes api server to take advantage of any new etcd settings that may have changed
+		// this only happens if the service is already running
+		status, err := getServiceStatus(APIServerServiceName)
```

Comment:
Should not this be `getServiceStatus(service)`? Ditto for the warning message below.